packages/pluggableWidgets/column-chart-native/src/components/ColumnChart.tsx

```diff
@@ -0,0 +1,328 @@
+    series: ColumnChartSeries[];
+    sortOrder: SortOrderEnum;
+    style: ColumnChartStyle;
+    presentation: string;
+    showLegend: boolean;
+    showLabels: boolean;
+    xAxisLabel?: string;
+    yAxisLabel?: string;
+    warningPrefix?: string;
+}
+
+export interface ColumnChartSeries {
+    dataPoints: ColumnDataPoints;
+    xFormatter?: (xValue: number | Date | string) => string;
+    yFormatter?: (yValue: number | Date | string) => string;
+    name?: string;
+    customColumnStyle?: string;
+}
+
+export type ColumnDataPoints =
+    | Array<ColumnDataPoint<number, number>>
+    | Array<ColumnDataPoint<number, Date>>
+    | Array<ColumnDataPoint<Date, number>>
+    | Array<ColumnDataPoint<Date, Date>>
+    | Array<ColumnDataPoint<string, string>>
+    | Array<ColumnDataPoint<string, number>>
+    | Array<ColumnDataPoint<string, Date>>
+    | Array<ColumnDataPoint<number, string>>
+    | Array<ColumnDataPoint<Date, string>>;
+
+export interface ColumnDataPoint<X extends number | Date | string, Y extends number | Date | string> {
+    x: X;
+    y: Y;
+}
+
+function sortSeriesDataPoints(series: ColumnChartSeries[], sortingOrder: string): ColumnChartSeries[] {
+    const keysSum: { [key: string]: number } = {};
+    series.forEach(({ dataPoints }) => {
+        dataPoints.forEach(({ x, y }) => {
+            const key = x.toString();
+            if (key in keysSum) {
+                keysSum[key] += Number(y);
+            } else {
+                keysSum[key] = Number(y);
+            }
+        });
+    });
+
+    return series.map(seriesItem => {
+        const dataPoints = seriesItem.dataPoints as Array<ColumnDataPoint<string, number>>;
+        seriesItem.dataPoints = Object.keys(keysSum)
+            .sort((key1, key2) => {
+                return sortingOrder === "descending" ? keysSum[key2] - keysSum[key1] : keysSum[key1] - keysSum[key2];
+            })
+            .map(key => ({
+                x: key,
+                y: dataPoints.find((point: ColumnDataPoint<string, number>) => point.x === key)?.y || 0
+            }));
+
+        return seriesItem;
+    });
+}
+
+export function ColumnChart({
+    name,
+    presentation,
+    series,
+    showLabels,
+    xAxisLabel,
+    yAxisLabel,
+    showLegend,
+    sortOrder,
+    style,
+    warningPrefix
+}: ColumnChartProps): ReactElement | null {
+    const sortedSeries = useMemo(() => {
+        const isNotSortable = series.some(({ dataPoints }) =>
+            dataPoints.some(({ x, y }) => typeof x !== "string" || typeof y !== "number")
+        );
+        return isNotSortable ? series : sortSeriesDataPoints(series, sortOrder);
+    }, [sortOrder, series]);
+
+    const [chartDimensions, setChartDimensions] = useState<{ height: number; width: number }>();
+
+    const warningMessagePrefix = useMemo(() => (warningPrefix ? warningPrefix + "i" : "I"), [warningPrefix]);
+
+    const dataTypesResult = useMemo(() => getDataTypes(sortedSeries), [sortedSeries]);
```

Comment:
It does not matter whether you do this on the sorted series or the original one, right? In that case, if you move this to the beginning, you already know the data types and don't have to iterate over the data points again to check if they are sortable.

Reply:
done